packages/ckb-sdk-core/src/index.ts

```diff
@@ -420,6 +420,49 @@ class CKB {
     }
   }
 
+  public calculateDaoMaximumWithdraw = async (
+      outPoint: CKBComponents.OutPoint,
+      withdrawBlockHash: string
+  ): Promise<string> => {
+    const depositTx = await this.rpc.getTransaction(outPoint.txHash)
+    if (depositTx.txStatus.status !== 'committed') throw new Error('Transaction is not committed yet')
+    const depositCell = depositTx.transaction.outputs[+outPoint.index]
+    const depositBlockHash = depositTx.txStatus.blockHash
+    const depositHeader = await this.rpc.getHeader(depositBlockHash)
+    const depositAr = this.#extracDaoData(depositHeader.dao).ar
+    const withDrawHeader = await this.rpc.getHeader(withdrawBlockHash)
+    const withdrawAr = this.#extracDaoData(withDrawHeader.dao).ar
+    const { JSBI } = this.utils
+    const occupiedCapacity = JSBI.asUintN(64, JSBI.BigInt(10200000000));
+    return JSBI.add(
```

Comment:
This algorithm can be placed in `utils` so that we can calculate the max withdraw capacity without states or network requests.